lib/addons/fluxcd/gitrepository.ts

```diff
@@ -0,0 +1,31 @@
+import { FluxCDAddOnProps } from '.';
+import { GitRepositoryReference } from '../../spi';
+import * as spi from "../../spi";
+
+/**
+ * Flux GitRepository API defines a Source to produce an Artifact for a Git repository revision.
+ */
+export class FluxGitRepository {
+
+    constructor(private readonly bootstrapRepo: spi.ApplicationRepository | undefined) {}
+
+    public generate(fluxcdAddonProps: FluxCDAddOnProps) {
+
+        const repository = fluxcdAddonProps.bootstrapRepo!;
+        return {
+            apiVersion: "source.toolkit.fluxcd.io/v1beta2",
+            kind: "GitRepository",
+            metadata: {
+                name: repository.name,
+                namespace: fluxcdAddonProps.namespace
+            },
+            spec: {
+                interval: "5m0s",
```

Comment:
is this value for sync internal something that customers may want to override? should we make it configurable?

Reply:
this also doesn't work with CSI secretprovider like the argo Addon works